internal/testutils/tls_creds.go

```diff
@@ -74,3 +74,25 @@ func CreateServerTLSCredentials(t *testing.T, clientAuth tls.ClientAuthType) cre
 		ClientCAs:    ca,
 	})
 }
+
+func CreateServerTLSCredentialsSPIFFE(t *testing.T, clientAuth tls.ClientAuthType) credentials.TransportCredentials {
```

Comment:
The name of this function is confusing: do you mean the server's certificates are SPIFFE certs? That the server is using SPIFFE verification? Both? Can you please make the function name more clear? :)